lib/mongoid/fields/localized.rb

```diff
@@ -86,7 +85,10 @@ def lookup(object)
         end
         return value unless value.nil?
         if fallbacks? && ::I18n.respond_to?(:fallbacks)
-          object[::I18n.fallbacks[locale].map(&:to_s).find{ |loc| object.has_key?(loc) }]
+          fallback_key = ::I18n.fallbacks[locale].find do |loc|
+            object.has_key?(loc.to_s) || object.has_key?(loc)
+          end
+          object[fallback_key.to_s] || object[fallback_key]
```

Comment:
What is the reason for stringified key being used in preference to the actual key?

Reply:
I'm not sure, but if I had to guess, I would think it's because rails does it this way. Just like they stringify the keys of hashes... I wouldn't be opposed to changing this to be a BSON document, but I also think this is fine as it is.

Reply:
If you don't have a strong reason can this be

```
object[fallback_key] || object[fallback_key.to_s]
```

Reply:
I rather not, and I'll tell you why. When adding values to the localized fields, they are added to the translations hash using string keys. The symbol keys would only be in the translations hash if users put them there themselves. It seems better to prioritize string keys for that reason, and that is what's done a few lines above this one (existing code).

Reply:
i18n keys are symbols, the localized field value hashes' keys are strings, therefore it makes some sense to check first for the stringified key and then use the original key.